src/main/java/org/fungover/mmotodo/service/AuthService.java

```diff
@@ -0,0 +1,151 @@
+package org.fungover.mmotodo.service;
+
+
+import jakarta.servlet.http.Cookie;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import org.fungover.mmotodo.dto.GithubUser;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.*;
+
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.annotation.AuthenticationPrincipal;
+
+
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.oauth2.client.OAuth2AuthorizedClient;
+import org.springframework.security.oauth2.client.OAuth2AuthorizedClientService;
+import org.springframework.security.oauth2.client.authentication.OAuth2AuthenticationToken;
+import org.springframework.security.oauth2.core.OAuth2AccessToken;
+import org.springframework.security.oauth2.core.user.OAuth2User;
+
+import org.springframework.stereotype.Service;
+
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.context.request.RequestContextHolder;
+import org.springframework.web.context.request.ServletRequestAttributes;
+
+
+import java.util.Base64;
+import java.util.Map;
+
+@Service
+public class AuthService {
+
+    @Value("${spring.security.oauth2.client.registration.github.clientId}")
+    private String clientId;
+
+    @Value("${spring.security.oauth2.client.registration.github.clientSecret}")
+    private String clientSecret;
+
+    @Value("${github.api.logout.url}")
+    private String githubDeleteUrl;
+
```

Comment:
Why is the logout url called githubDeleteUrl?

Reply:
I named it that way because it revokes/deletes the user from the OAuth app. However, a more appropriate name would be githubSignOutUrl.